### PR TITLE
Update JustGiving::Configuration::BASE_URI to a class method

### PR DESCRIPTION
**Pull request by robbiecloset**
- Added JustGiving::Configuration.base_uri to be an env-depdendent class method.
- Update faraday & co gems to be latest versions, and updated the connection creation. 
- Test cleanup.

### CODE REVIEW

**Comment by eellson** [comment]
:thumbsup: this looks good to me (plus we know it works as it's being used on a real life production app right now!). Might be worth someone else giving it some eyes too tho

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+ruby-1.9.3-p194
```
Comment (philnash) [inline .ruby-version]: Probably shouldn't be checking this, or `.ruby-gemset`, into the gem.

**Comment by philnash** [comment]
Looks good, other than some unnecessary development files in the repo. :shipit:

**Comment by robbiecloset** [comment]
Thanks @eellson & @philnash. Good call on the the .ruby-\* files; they've been removed now and ignored.

**Comment by robbiecloset** [comment]
@bob-p, mind if I bump the gem? Or is that something you'll need to do?

**Comment by bob-p** [comment]
No, think you can do so @johnideal I can add you to the rubygems if you're not already and you can push the new version.

**Comment by bob-p** [comment]
You should be an owner of that gem now @johnideal

**Pull request merged**